proposals/0312-create-account-prefunded.md

```diff
@@ -12,68 +12,118 @@ feature: create-account-prefunded
-own with 2-3 CPI calls: sometimes `Transfer`, then `Allocate`, and then
-`Assign`. While these actions themselves are minimally expensive, the overhead
-incurred with every Cross-Program Invocation - depth check, signer check,
+than forcing the fee payer of the transaction to provide all of the required
+lamports. In this and similar instances, developers currently must manually
+construct a patched `CreateAccount` call of their own with 2-3 CPI calls:
+sometimes `Transfer`, then `Allocate`, and then `Assign`. While these
+actions themselves are minimally expensive, the overhead incurred
+with every Cross-Program Invocation - depth check, signer check,
 account copy, etc. - can make up the bulk of the computation done in the
-transaction.
+transaction. Each CPI incurs a minimum of 1_000 compute units, plus
+additional amounts depending on the instruction and account length.
 
-`CreateAccountPrefunded` performs `allocate`, `assign`, and `transfer`
+`CreatePrefundedAccount` performs `allocate`, `assign`, and `transfer`
 without asserting that the created account has zero lamports. Applications
 which do not need to `transfer` can specify 0 lamports, effectively providing
 them with an `AllocateAndAssign`.
 
+p-ATA program benchmarks demonstrate that use of `CreatePrefundedAccount`
+can save approximately 2_500 compute units:
+https://github.com/solana-program/associated-token-account/pull/102
+
 This is a stopgap measure, as it extends the undesired interface sprawl of
 helper instructions such as `CreateAccount`. However, any redesign of
 Cross-Program Invocations to safely reduce overhead would be an extensive
 protocol upgrade and slow to land. In the meantime, the network will benefit
-from `CreateAccountPrefunded`.
+from `CreatePrefundedAccount`.
 
 ## New Terminology
 
-* Prefunded: an account which receives lamports to pay for its rent in whole
-or in part before its space is `allocate`d and its owner is `assign`ed.
+* Prefunded Account: an account which receives lamports to pay for its rent
+in whole or in part before its space is `allocate`d and its owner is
+`assign`ed.
 
 ## Detailed Design
 
-`create_account_prefunded()` takes the same arguments as `create_account()`
-and performs the same operations; however, it does not fail if the new
-account's `lamports > 0`.
+`CreatePrefundedAccount` is added as a system program instruction, identical
+to `CreateAccount` in all but discriminant (13).
+
+### Instruction Data
+
+| Field | Type | Description |
+|---|---|---|
+| Discriminant | `u32` | The instruction discriminant, value `13`. |
+| `lamports` | `u64` | Number of lamports to transfer to the new account. |
+| `space` | `u64` | Number of bytes of memory to allocate. |
+| `owner` | `Pubkey` | Address of program that will own the new account. |
+
+### Accounts
+
+The instruction requires the following accounts:
+
+| Index | Role | Description |
+|---|---|---|
+| 0 | `[WRITE, SIGNER]` | **Funding account**: The account that will pay
```

Comment:
What do you think about making this account optional? This would make it different from `CreateAccount`, but that should be ok